src/lamp_py/migrations/versions/performance_manager_dev/008_668d8e124504_redo_dwell_time_update.py

```diff
@@ -0,0 +1,191 @@
+                        FROM
+                            static_stop_times
+                        JOIN static_trips ON
+                            static_stop_times.static_version_key = static_trips.static_version_key
+                            AND static_stop_times.trip_id = static_trips.trip_id
+                        JOIN static_stops ON
+                            static_stop_times.static_version_key = static_stops.static_version_key
+                            AND static_stop_times.stop_id = static_stops.stop_id
+                        JOIN static_service_id_lookup ON
+                            static_stop_times.static_version_key = static_service_id_lookup.static_version_key
+                            AND static_trips.service_id = static_service_id_lookup.service_id
+                            AND static_trips.route_id = static_service_id_lookup.route_id
+                        JOIN static_routes ON
+                            static_stop_times.static_version_key = static_routes.static_version_key
+                            AND static_trips.route_id = static_routes.route_id
+                        WHERE
+                            static_stop_times.static_version_key = {static_version_key}
+                            AND static_trips.static_version_key = {static_version_key}
+                            AND static_stops.static_version_key = {static_version_key}
+                            AND static_service_id_lookup.static_version_key = {static_version_key}
+                            AND static_routes.static_version_key = {static_version_key}
+                            AND static_routes.route_type != 3
+                            AND static_service_id_lookup.service_date = {service_date}) AS static_trips_sub ON
+                        rt_trips_sub.static_trip_id_guess = static_trips_sub.static_trip_id
+                        AND rt_trips_sub.static_version_key = static_trips_sub.static_version_key
+                        AND rt_trips_sub.parent_station = static_trips_sub.parent_station
+                        AND rt_trips_sub.rt_trip_stop_rank >= static_trips_sub.static_trip_stop_rank
+                    ORDER BY
+                        rt_trips_sub.service_date,
+                        rt_trips_sub.pm_trip_id,
+                        rt_trips_sub.parent_station,
+                        static_trips_sub.static_trip_stop_rank) AS trip_for_metrics
+                WHERE
+                    trip_for_metrics.first_stop_flag = FALSE
+                    AND trip_for_metrics.stop_count > 1) AS t_dwell_times
+            WHERE
+                vehicle_events.pm_trip_id = t_dwell_times.pm_trip_id
+                AND vehicle_events.service_date = t_dwell_times.service_date
+                AND vehicle_events.parent_station = t_dwell_times.parent_station
+                AND (t_dwell_times.dwell_time_seconds IS NULL OR t_dwell_times.dwell_time_seconds >= 0)
```

Comment:
This updated `AND` clause will update dwell_time values that are NULL

Reply:
ah. thats a subtle distinction.